compact_str/src/repr/boxed/mod.rs

```diff
@@ -407,6 +410,32 @@ impl BoxString {
             }
         }
     }
+
+    /// Tries to reallocate the buffer, and sets the new capacity on success.
+    ///
+    /// Returns the new capacity as `Ok(…)` if the string was reallocated.
+    /// Or the old capacity as `Err(…)` if the string could not be reallocated.
+    pub unsafe fn realloc(&mut self, new_capacity: usize) -> Result<usize, usize> {
+        let old_capacity = self.capacity();
```

Comment:
if you call `self.cap.as_usize()` that will return an error when the capacity is on the heap. For now it would be totally fine with me to just `return Err(...)` in that case to avoid having to handle the change in layouts